### ISSUE TITLE
fb-sdk not working with iron-router

### ISSUE THREAD

**Issue by rnarayan** [hugesuccess/facebook-sdk#4]
Hi Victor,

Thanks a lot for this fb-sdk package. It's great. 
I used your package to successfully connect with facebook in my test app. My test app doesn't use iron-router.

However, when I use iron-router, it doesn't work. 
I have read your documentation thoroughly and I have added the following code(with my template name). But I get the error - "FB" is not defined.

Template.<template-name>.rendered = function() {
    try {
        FB.XFBML.parse();
    }catch(e) {}  
};

I hope you can help.  I have the "fbAsyncInit" file in my client directory. 

Thanks.

**Comment by hugesuccess** [comment]
Is it deployed? Facebook doesn't work on localhost.

**Comment by rnarayan** [comment]
My test app where I don't use iron router, worked with facebook-sdk on localhost:3000. I was able to connect and download Likes of logged-in user. However, when I created another app with iron-router, it stopped working even though I did put the FB.XFBML code in the template render as mentioned by you. 

I will try to deploy and test it. Meanwhile, any other suggestion would be helpful.

Here is my code that works. This test app has only one package installed - facebook-sdk.

In the client directory. (client/fbinit.js)
window.fbAsyncInit = function() {
    FB.init({
      appId      : 'XXXXXXXXXXX',  //real id in the app
      status     : true,
      xfbml      : true
    });
  };

<p>client/fbsdk.js</b>
 Template.hello.events({
    'click #facebook' : function () {  
      event.preventDefault();
      var url;
      var scope =  {scope: 'email,user_likes'};
      var Likes = [];

```
   FB.login (fbLogin, scope);

    function fbLogin(response) {
      if (response.authResponse) {
        FB.api('/me', getUserProfile );
        getFBlikes();          
      } else {
        console.log('User cancelled login or did not fully authorize.');
      }
   }

    function getUserProfile(response) {
        console.log(response.name);
    }  

    function getFBlikes() {
      FB.api("/me/likes", function(likes) {
        saveLikes(likes);           
      });
    } 

   function saveLikes(likes){
    if (likes && !likes.error) {
     console.log(likes); 
     $.each(likes.data, function(key,val){
      var like = {};
      like.c = val.category;
      like.id = val.id;
      like.n = val.name;
      Likes.push(like);  
     });        
    }
   }
}
```

  });

**Comment by hugesuccess** [comment]
I would have to see the code that's not working. If you can upload the non working code to a gist I'll look at it.

**Comment by rnarayan** [comment]
Hi Victor,

I found the issue. facebook-sdk package was not installed. There must have been an error message when I tried to install it the first time which I didn't read. I had assumed it was installed correctly. What could go wrong? Perhaps a typo.
I installed it correctly this time and the code works like a charm. The code in my development app is the same as what I pasted above, with the added difference that I have iron-router installed.

I am sorry to have wasted your time. If its any consolation, I ended up wasting a lot of my own time and frustrated while trying to find the error. Next time I will remember to read the system messages carefully.

Thanks a lot. This package has been very valuable to us.
